### PR TITLE
Feat(type-designer): persist data to scd file

### LINKED ISSUE #87: Persist new data added to scd file following a given data structure (xml template)
# Story: Persist new data added to scd file following a given data structure

**As** a User
**I want to** persist the changes made by adding new element to an SCD file
**So that** I can download the updated SCD file, and rework it later on needs.

## Acceptance Criteria
- [x] The XML structure should be updated when elements are added or moved, based on #121 
- [x] The downloaded XML file should contain the updated hierarchical structure.

This has been deprioritized.
- [ ] #136

### PR DESCRIPTION
**Pull request by michelguerin**
Closes #87 

@illia-solovei-s1 commit history should described the work done in a proper order.
The most interested part is here :  24f9645

### CODE REVIEW

**Comment by michelguerin** [comment]
@fwagnerSE & @salvar3nga you might be interested to look at those commits : b80516d & 4653440
I added the editCount props from the OpenSCD Instance that was already used by our legacy plugins to trigger update made on the document (it is also used to keep trace of the history on the instance)

File: packages/core/src/constants/element.constant.ts [hunk 1]
```diff
@@ -1,8 +1,118 @@
...
-	ied: 'IED',
-	lDevice: 'Logical Device',
-	lNode: 'Logical Node'
+export const BASE_STANDARD_ATTRIBUTES = ['id', 'desc', 'name'] as const
+
+export const SCD_ELEMENTS = {
+	substation: {
+		element: {
+			tag: 'Substation',
+			name: 'Substation',
```
Comment (salvar3nga) [inline packages/core/src/constants/element.constant.ts:7]: To utilize TS we can think about creating types or interfaces for the SCD_Elements. Thoughts?

File: packages/plugins/type-designer/src/components/content-card.svelte [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+//====== INITIALIZATION ======//
+
+// props
+export let name: string
+export let elementType: BaseElementStandardAttributes
+
+//====== FUNCTIONS ======//
+
+function handleCardClick(elementType: BaseElementStandardAttributes) {
+	console.log('Clicked:', elementType)
```
Comment (salvar3nga) [inline packages/plugins/type-designer/src/components/content-card.svelte:15]: console.log

File: packages/core/src/constants/element.constant.ts (hunk 1 above)
Comment (illia-solovei-s1) [reply packages/core/src/constants/element.constant.ts:7]: It already exists and defined here:
`/scd-queries/data-type-templates/types.data-type-templates.d.ts`: `TypeElements`

Comment (michelguerin) [reply packages/core/src/constants/element.constant.ts:7]: I wrote this part as the source of truth of our plugins, and is seen by TS as a constant so ReadOnly properties. I use this to create our types (so it kind of works the other way around 🙃 )

File: packages/plugins/type-designer/src/stores/data-types-templates.store.ts [hunk 3]
```diff
@@ -1,43 +1,161 @@
...
+
+	const newElementId = `${SCD_ELEMENTS[columnKey].type.baseId}${iterateType}`
+	const newElementName = `${SCD_ELEMENTS[columnKey].type.baseName}${iterateType}`
+
+	const insertElementBefore = SCD_ELEMENTS[columnKey].type.insertBefore
+	const firstElementOfNextType = insertElementBefore
+		? get(typeElements)[insertElementBefore][0]?.element
+		: undefined
+
+	const newElement = pluginStore.addElementToXmlDocument({
```
Comment (illia-solovei-s1) [inline packages/plugins/type-designer/src/stores/data-types-templates.store.ts:129]: newElement is never used, you can just leave the function execution.

File: packages/core/src/constants/element.constant.ts (hunk 1 above)
Comment (illia-solovei-s1) [reply packages/core/src/constants/element.constant.ts:7]: Solved

**Review by illia-solovei-s1 — APPROVED** [review]

File: packages/plugins/type-designer/src/stores/data-types-templates.store.ts (hunk 3 above)
Comment (michelguerin) [reply packages/plugins/type-designer/src/stores/data-types-templates.store.ts]: done

**Pull request merged**